lib/Froxlor/Cron/Http/Apache.php

```diff
@@ -241,6 +241,7 @@ public function createIpPort()
 						$this->virtualhosts_data[$vhosts_filename] .= '  FcgidIdleTimeout ' . Settings::Get('system.mod_fcgid_idle_timeout') . "\n";
 						if ((int) Settings::Get('system.mod_fcgid_wrapper') == 0) {
 							$this->virtualhosts_data[$vhosts_filename] .= '  SuexecUserGroup "' . Settings::Get('system.mod_fcgid_httpuser') . '" "' . Settings::Get('system.mod_fcgid_httpgroup') . '"' . "\n";
+							$this->virtualhosts_data[$vhosts_filename] .= '  SetEnvIf Authorization .+ HTTP_AUTHORIZATION=$0' . "\n";
```

Comment:
is there an equivilant (needed) for Nginx and Lighttpd?

Reply:
AFAIR only apache strips the `Authorization` header when it contains an unknown scheme, so nginx and lighttpd should be unaffected.

Reply:
Good question, I've checked all the possible configurations to see if HTTP authentication works. As long as FCGID and PHP-FPM are not used for the Froxlor virtual host everything is fine. If they are used then:

* Apache 2.4 with fcgi needs the suggested change
* Apache 2.4 with PHP-FPM only works if the PHP configuration for the Froxlor VHost has this setting enabled:
> Add "-pass-header Authorization" / "CGIPassAuth On" to vhosts:
* nginx with PHP-FPM works because nginx seems to always pass the HTTP_AUTHORIZATION header to the fastcgi script.
* Nginx with a separate PHP backend works for the same reason, it forwards the header.
* LigHTTPd with neither FCGID nor PHP-FPM gives a configuration that can't do anything with PHP by my understanding. I've ignored this case.
* LigHTTPd with FCGID doesn't support FCGID for the Froxlor vHost according to the UI. I've ignored this case as well.
* LigHTTPd with PHP-FPM also seems to always pass the HTTP_AUTHORIZATION header to the fastcgi script.

Should I somehow make sure that "-pass-header Authorization" / "CGIPassAuth On" is always enabled for the PHP configuration of Froxlors virtual host?

Reply:
IMO apache 2.4+php-fpm also works without `Add "-pass-header Authorization"` and `CGIPassAuth On`.
I simply have a rewrite rule for this that I stole from Symfony, so far this worked well for any apache setup:
```
    # Sets the HTTP_AUTHORIZATION header removed by Apache
    RewriteCond %{HTTP:Authorization} .
    RewriteRule ^ - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
```
I do not know if this is a proper substitution for your `SetEnvIf` in all cases but I guess so.